File: LB-Assignments/Assignment_9/program4.c
#include<stdio.h>

int MultDigits(int iNo)
{
    int iMult = 1;
    int iRem = 0;

    while(iNo != 0)
    {
        iRem = iNo % 10;

        if(iRem != 0)
        {
            iMult = iMult * iRem;
        }

        iNo = iNo/ 10;
    }

    return iMult;

}
int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = MultDigits(iValue);

    printf("%d",iRet);
    
    return 0;
}
// Time Complexity : O(N)

File: LB-Assignments/Assignment_18/program4.java
/*
write a program to print sum of even digits and odd digits separately
*/
import java.util.*;

class Logic 
{
    void sumEvenOddDigits(int iNo)
    {
        int iSumEven = 0, iSumOdd = 0, iRem = 0;

        while(iNo != 0)
        {
            iRem = iNo % 10;
            if(iRem % 2 == 0)
            {
                iSumEven += iRem;
            }
            else
            {
                iSumOdd += iRem;
            }
            iNo = iNo /10;
        }
        System.out.println("Sum of even digits : "+iSumEven);
        System.out.println("Sum of odd digits : "+iSumOdd);
    }
}

class program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        
        int iValue = 0;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        Logic lobj = new Logic();

        lobj.sumEvenOddDigits(iValue);
        
    }
}
